qiskit/qobj/qasm_qobj.py

```diff
@@ -368,7 +368,32 @@ class QasmQobjExperimentHeader(QobjDictField):
 
 class QasmQobjExperimentConfig(QobjDictField):
     """Configuration for a single QASM experiment in the qobj."""
-    pass
+
+    def __init__(self, calibrations=None, **kwargs):
+        """
+        Args:
+            calibrations (QasmQobjCalibrations): Information required for
+                                                 Pulse gates.
+            kwargs: Additional free form key value fields to add to the
+                configuration.
+        """
+        if calibrations:
+            self.calibrations = calibrations
+        super().__init__(**kwargs)
+
+
+class QasmQobjCalibrations(QobjDictField):
```

Comment:
If this class doesn't need to handle arbitrary key value pairs as kwarg inputs and need to be attribute accesible than this shouldn't be a subclass of `QobjDictField`. `QobjDictField` is only used for things were we allow any KV pair to be passed in and have that treated like a defined attribute, it's basically a backwards compat thing for how marshmallow was configured.

This will have issues with things like deepcopy and pickling because it doesn't allow `**kwargs` and has positional params defined. This should probably just be a flat class that defines the attributes in it.